sbt-scala-native-bindgen/src/main/scala/org/scalanative/bindgen/sbt/ScalaNativeBindgenPlugin.scala

```diff
@@ -126,34 +117,26 @@ object ScalaNativeBindgenPlugin extends AutoPlugin {
         sourceGenerators += Def.task { nativeBindgen.value },
         target in nativeBindgen := sourceManaged.value / "sbt-scala-native-bindgen",
         nativeBindgen := {
-          val bindgenPath     = nativeBindgenPath.value
-          val bindings        = nativeBindings.value
+          val bindgen         = Bindgen(nativeBindgenPath.value)
+          val optionsList     = nativeBindings.value
           val outputDirectory = (target in nativeBindgen).value
           val logger          = streams.value.log
 
-          bindings.map {
-            binding =>
-              val output = outputDirectory / s"${binding.name}.scala"
-              val result = Bindgen()
-                .bindgenExecutable(bindgenPath)
-                .header(binding.header)
-                .name(binding.name)
-                .maybe(binding.link, _.link)
-                .maybe(binding.packageName, _.packageName)
-                .maybe(binding.excludePrefix, _.excludePrefix)
-                .generate()
+          // FIXME: Check uniqueness of names.
 
-              result match {
+          optionsList.map {
+            case options: BindingOptions.Impl =>
```

Comment:
Is it possible to have an instance of any other type? `BindingOptions` is `sealed trait` and there is only one implementation

Reply:
No, it should not be possible. Actually, the `case` is not needed here.